modules/cache/main.tf

```diff
@@ -0,0 +1,40 @@
+#----------------------------
+# create GCS bucket for cache
+#-----------------------------
+resource "google_storage_bucket" "cache" {
+  name          = var.bucket_name
```

Comment:
Generate bucket name, if not passed. Default should be `null`. If not passed, default should be `gitlab-runner-cache-<random-suffix>`